client-app/app/controllers/index.js

```diff
@@ -171,4 +182,70 @@ export default class IndexController extends Controller {
 
     debounce(this, this.doSearch, term, 250);
   }
+
+  @action
+  toggleGroupingPatternFromSelectedRows() {
+    this.buildingGroupingPattern = !this.buildingGroupingPattern;
+    this.rowMessagesForGroupingPattern = [];
+  }
+
+  @action
+  async createGroupingPatternFromSelectedRows() {
+    if (this.rowMessagesForGroupingPattern.length < 2) {
+      // eslint-disable-next-line no-alert
+      return alert(
+        "You must select at least 2 rows to create a grouping pattern"
+      );
```

Comment:
Mentioned this in chat, but you won't need this if the "Create grouping pattern" button is only shown when there are at least 2 checked items.